Task_1.py

```diff
@@ -0,0 +1,160 @@
+    comment = ['messages with the most “likes” from the entire data set and the application (asin) associated with it',
+               ';\n like =', main_dict[sort_likes[0][0]]['helpful'][0],
+               ';\n asin:', main_dict[sort_likes[0][0]]['asin'],
+               ';\n reviewText:', main_dict[sort_likes[0][0]]['reviewText']
+               ]
+    return comment
+
+# 1) Task_1.py: to create file general-stats.cvs containing information about:
+#    1.3. the shortest interval between ratings of one user (among all users) and the length of both messages which create this interval;
+def nearest_review(main_dict):
+    all_timereview = {}
+    for key, value in main_dict.items():
+        if 'reviewerName' in value.keys() and 'unixReviewTime' in value.keys() and value['reviewerName'] != 'Amazon Customer':
+            if all_timereview.get(value['reviewerName']) == None:
+                all_timereview[value['reviewerName']] = {}
+                all_timereview.get(value['reviewerName']).setdefault(key, value['unixReviewTime'])
+            else:
+                all_timereview.get(value['reviewerName']).setdefault(key, value['unixReviewTime'])
+    deltatime = []
+    for item in all_timereview.values():
+        if len(item) > 1:
+            sort = sorted(item.items(), key=itemgetter(1))
+            for i in range(1, len(sort)):
+                a = i-1
+                deltatime.append([sort[i][0], sort[a][0], sort[i][1] - sort[a][1]])
+    time_deltatime = []
+    global analyzed_data
+    for message in deltatime:
+        if message[2] != 0:
+            time_deltatime.append(message[2])
+            analyzed_data.append(message[0])
+            analyzed_data.append(message[1])
+    for message in deltatime:
+        if message[2] == min(time_deltatime):
+            comment = [
+                'the shortest interval between ratings of one user (among all users) and the length of both messages which create this interval: \n',
+                'interval = ' + str(message[2]//60//60) + 'hour' + str(message[2]//60%60) + 'min'+ str(message[2]%60%60) + 'sec;\n',
+                'lenght comment_1: ' + str(len(main_dict[message[0]]['reviewText'])) + ';\n',
+                'lenght comment_2: ' + str(len(main_dict[message[1]]['reviewText']))
+            ]
+            if comment:
+                break
+    return comment
+
+
+# 1) Task_1.py: to create file general-stats.cvs containing information about:
+#    1.4. the application which received the most useless message;
+def bad_comment(main_dict):
+    helpfulness = []
+    for key, value in main_dict.items():
+        if value['helpful'][1] != 0:
+            helpfulness.append([key, value['helpful'][0]/value['helpful'][1]])
+    result_helpfulness = []
+    for message in helpfulness:
+        result_helpfulness.append(message[1])
+    for message in helpfulness:
+        if message[1] == min(result_helpfulness):
```

Comment:
You can use min func directly, just pass named parameter _key_ to compare to reviews. Please see the comment above regarding max function.